src/main/java/games/battlelore/components/Terrain.java

```diff
@@ -0,0 +1,16 @@
+package games.battlelore.components;
+
+import core.components.Component;
+import utilities.Utils;
+
+public class Terrain extends Component {
+
+    public Terrain(Utils.ComponentType type, String name) {
+        super(type, name);
+    }
+
+    @Override
+    public Component copy() {
+        return null;
```

Comment:
You likely don't need this class at all, since it doesn't contain anything that Component doesn't - either way, returning null for copies is not indicated.